openquakeplatform/openquakeplatform/static/js/irv_viewer.js

```diff
@@ -14,1043 +14,850 @@
+        }
     }
 
-    function processIndicators(e) {
-        var districName = e.data.name_1;
-        //$('#pcp-charts').dialog('open');
-        // Get the SVIR data from the utfGrid
-        var tmpIri = {};
-        var tmpPI;
-        var tempCI;
-        if (e.data) {
-            // Find the variables that are primary indicators and their respecive category
-            var ci = 'categoryIndicator';
-            findCategoryIndicators(pdData, [ci]);
-            var pi = 'primaryIndicator';
-            findPrimaryIndicators(pdData, [pi]);
-            var svi = 'svi';
-            findSvi(pdData, [svi]);
-
-            // Create an object for each of the category indicators
-            for (var n = 0; n < pdTempCategoryIndicator.length; n++) {
-                parentChildKey[pdTempCategoryIndicator[n]] = [];
-                var category = pdTempCategoryIndicator[n];
-            }
-            var iri = e.data.ir;
-            iri = iri.split(',');
-            var munic = e.data.municipio;
-            munic = munic.split(',');
-            var pri = e.data.pri;
-
-            if (pri == undefined) {
-                pri = e.data.aal;
-            }
-            var munic_num = e.data['municipio'].split(',').length;
-            municipality = e.data['municipio'].split(',');
-
+    /////////////////////////////////////////
+    //// Compute the Category indicators ////
+    /////////////////////////////////////////
 
-            /////////////////////////////////////////////
-            //// Create the primary indicator objects ///
-            /////////////////////////////////////////////
+    //build the category indicator object
+    var catData = [];
+    tempString = [];
 
-            for (var k in weightedPrimaryIndicator) {
-                delete weightedPrimaryIndicator[k];
+    function generateObject(temp) {
+        var munic = temp[0];
+        var theme = temp[1];
+        var value = parseFloat(temp[2]);
+        // add the theme and value to each category data object
+        for (var i = 0; i < catData.length; i++) {
+            if (catData[i].region == munic) {
+                catData[i][theme] = value;
             }
+        }
+    }
 
-            for (var k in sessionPrimaryIndicator) {
-                delete sessionPrimaryIndicator[k];
-            }
+    // setup catData with all the municipalities
+    var la = layerAttributes.features;
+    for (var s = 0; s < la.length; s++) {
+        var temp = {};
+        temp.region = la[s].properties[selectedRegion];
+        catData.push(temp);
+    }
 
-            for (var k in primaryIndicator) {
-                delete primaryIndicator[k];
-            }
+    // Find the theme information
+    for (var m = 0; m < socialVulnIndex.length; m++) {
+        var operator = socialVulnIndex[m].operator;
+        var weight = socialVulnIndex[m].weight;
+        var name = socialVulnIndex[m].name;
```

Comment:
(I think in this case `name` is correct, because themes aren't stored in actual fields in the layer)